remoting/src/main/java/org/apache/rocketmq/remoting/protocol/RocketMQSerializable.java

```diff
@@ -149,6 +150,9 @@ public static RemotingCommand rocketMQProtocolDecode(final byte[] headerArray) {
         // String remark
         int remarkLength = headerBuffer.getInt();
         if (remarkLength > 0) {
+            if (remarkLength > headerArray.length) {
```

Comment:
Will it cause the similar problem if you modify other values in the header?

Reply:
> Will it cause the similar problem if you modify other values in the header?

In addition to remarkLength and extFieldsLength, modifying other values in the header will not create large objects, but will only fail to read and throw exceptions